stream_alert/alert_processor/outputs/pagerduty.py

```diff
@@ -479,8 +592,38 @@ def dispatch(self, **kwargs):
         incidents_url = self._get_endpoint(self._base_url, self.INCIDENTS_ENDPOINT)
 
         try:
-            success = self._post_request_retry(incidents_url, incident, self._headers, True)
+            incident = self._post_request_retry(incidents_url, incident_data, self._headers, True)
         except OutputRequestFailure:
-            success = False
+            incident = False
 
-        return self._log_status(success)
+        if not incident:
+            LOGGER.error('Could not create main incident, %s', self.__service__)
+            return self._log_status(False)
+
+        # Extract the incident id from the incident that was just created
+        incident_id = incident.get('incident').get('id')
+
+        # Create alert to hold all the incident details
+        event_data = events_v2_data(creds['integration_key'], **kwargs)
+        event = self._create_event(event_data)
+        if not event:
+            LOGGER.error('Could not create incident event, %s', self.__service__)
+            return self._log_status(False)
+
+        # Lookup the incident_key returned as dedup_key to get the incident id
+        incident_key = event.get('dedup_key')
+
+        # Keep that id to be merged later with the created incident
+        event_incident_id = self._get_event_incident_id(incident_key)
+
+        # Merge the incident with the event, so we can have a rich context incident
+        # assigned to a specific person, which the PagerDuty REST API v2 does not allow
+        merging_url = '{}/{}'.format(incidents_url, incident_id)
+
+        merged = self._merge_incidents(merging_url, event_incident_id)
+
+        if not merged:
```

Comment:
I think this if block is safe to leave out, since essentially the same thing will happen just below on:

`return self._log_status(merged)`

And there is a logger message in `_log_status` that does something similar to what you're logging here.